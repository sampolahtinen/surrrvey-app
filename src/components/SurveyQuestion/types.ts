export interface SurveyQuestionProps {
  title: string;
  options: string[];
  onClick: (param: string) => void;
  selectedOption: string;
}
