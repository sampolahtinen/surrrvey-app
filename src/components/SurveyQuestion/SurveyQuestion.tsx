import React from "react";
import {
  QuestionsContainer,
  QuestionTitle,
  QuestionList,
  Option
} from "./SurveyQuestion.styles";
import { SurveyQuestionProps } from "./types";

const SurveyQuestion: React.FC<SurveyQuestionProps> = ({
  options,
  selectedOption,
  title,
  onOptionSelect
}) => {
  return (
    <QuestionsContainer style={{ marginBottom: "3.2rem" }}>
      <QuestionTitle>{title}</QuestionTitle>
      <QuestionList>
        {options.map((option: string) => (
          <Option
            key={title + option}
            onClick={() => onOptionSelect(option)}
            selected={selectedOption === option}
          >
            {option}
          </Option>
        ))}
      </QuestionList>
    </QuestionsContainer>
  );
};

export default SurveyQuestion;
