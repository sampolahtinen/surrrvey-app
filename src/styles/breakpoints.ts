export const breakPoints = {
  phone: "375px",
  tablet: "768px",
  laptop: "1440px"
};
