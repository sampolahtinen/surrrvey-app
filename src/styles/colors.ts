export const colors = {
  white: "white",
  offWhite: "#eeeeee",
  grey: "#323639",
  darkGrey: "#1f2024",
  black: "black",
  red: "#f44336",
  green: "#1ed36f",
  lila: "#6e1fff",
  darkLila: "#4f2848"
};
