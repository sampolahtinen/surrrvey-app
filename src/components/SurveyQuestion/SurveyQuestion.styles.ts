import styled from "styled-components";
import { colors } from "../../styles/colors";

export const QuestionList = styled.ul``;

export const QuestionTitle = styled.h3`
  margin-bottom: 2.4rem;
`;

export const QuestionsContainer = styled.div`
  width: 400px;
  height: 300px;
`;

export const Option = styled.li<{ selected: boolean }>`
  margin-bottom: 1.6rem;
  border-radius: 12px;
  background-color: ${props =>
    props.selected ? colors.darkLila : colors.offWhite};
  color: ${props => (props.selected ? colors.offWhite : colors.darkGrey)};
  padding: 0.8rem;
  &:hover {
    background-color: ${colors.darkLila};
    color: ${colors.offWhite};
  }
`;
