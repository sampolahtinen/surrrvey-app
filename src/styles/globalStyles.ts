import { createGlobalStyle } from "styled-components";
import { colors } from "./colors";

export const GlobalStyle = createGlobalStyle`
  @font-face {
  font-family: "GothamRounded";
  src: url("../static/fonts/gothamrounded-medium.otf") format("otf");
  font-display: fallback;
  @import url('https://fonts.googleapis.com/css?family=VT323&display=swap');
  @import url('https://fonts.googleapis.com/css?family=Roboto&display=swap');
  @import url('https://fonts.googleapis.com/css?family=Varela&display=swap');
  }
* {
  padding: 0;
  margin: 0;
  box-sizing: border-box;
  outline: none;
  font-family: "GothamRounded", "Calibre", sans-serif;
}

html {
  font-size: 62.5%;
  overflow-x: hidden;
}

body {
  height: 100%;
  width: 100%;
  font-size: 1.6rem;
  color: ${colors.darkGrey};
  background-attachment: fixed;
  background-color: white;
  font-family: 'Roboto', sans-serif;
  background-image: linear-gradient(to top, #dad4ec 0%, #dad4ec 1%, #f3e7e9 100%);
}

ul,
li {
  list-style: none;
}

a {
  text-decoration: none;
}

input {
  border: 0;
}

@media screen and (max-width: 420px) {
  html {
    font-size: 56.25%; /* 55% of 16px is 9px*/
  }
}

@media screen and (max-width: 768px) {
  html {
    font-size: 56.25%; /* 55% of 16px is 9px*/
  }
}
`;
