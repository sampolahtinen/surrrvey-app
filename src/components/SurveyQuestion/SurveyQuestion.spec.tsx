import React from "react";
import { shallow } from "enzyme";
import SurveyQuestion from "./SurveyQuestion";
import { Option, QuestionTitle } from "./SurveyQuestion.styles";

describe("SurveyQuestion Component", () => {
  const mockFunc = jest.fn();
  const wrapper = shallow(
    <SurveyQuestion
      title="testTitle"
      options={["test1", "test2"]}
      onOptionSelect={mockFunc}
      selectedOption={"test1"}
    />
  );

  it("renders", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("displays two options", () => {
    expect(wrapper.find(Option)).toHaveLength(2);
  });

  it("displays title", () => {
    expect(wrapper.find(QuestionTitle).text()).toEqual("testTitle");
  });

  it("handles option selection", () => {
    expect(
      wrapper
        .find(Option)
        .first()
        .prop("selected")
    ).toBeTruthy();
  });

  it("handles clicking", () => {
    wrapper
      .find(Option)
      .first()
      .simulate("click");
    expect(mockFunc).toHaveBeenCalled();
  });
});

// Need to test styled component, because its style is adapted based on props
describe("Question Component", () => {
  it("renders", () => {
    const wrapper = shallow(<Option selected={false} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders correctly when selected", () => {
    const wrapper = shallow(<Option selected={true} />);
    expect(wrapper).toMatchSnapshot();
  });
});
