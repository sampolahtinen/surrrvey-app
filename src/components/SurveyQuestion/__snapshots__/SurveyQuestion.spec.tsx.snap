// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question Component renders 1`] = `
<li
  className="sc-AykKF irIltD"
  selected={false}
/>
`;

exports[`Question Component renders correctly when selected 1`] = `
<li
  className="sc-AykKF bRvOzM"
  selected={true}
/>
`;

exports[`SurveyQuestion Component renders 1`] = `
<styled.div
  style={
    Object {
      "marginBottom": "3.2rem",
    }
  }
>
  <styled.h3>
    testTitle
  </styled.h3>
  <styled.ul>
    <styled.li
      key="testTitletest1"
      onClick={[Function]}
      selected={true}
    >
      test1
    </styled.li>
    <styled.li
      key="testTitletest2"
      onClick={[Function]}
      selected={false}
    >
      test2
    </styled.li>
  </styled.ul>
</styled.div>
`;
